sigstore/models.py

```diff
@@ -247,12 +249,17 @@ def _to_dict_rekor(self, is_message_signature: bool) -> dict[str, Any]:
         )
 
         # Fill in the appropriate kind
-        if is_message_signature:
-            tlog_entry.kind_version = rekor_v1.KindVersion(
-                kind="hashedrekord", version="0.0.1"
-            )
-        else:
-            tlog_entry.kind_version = rekor_v1.KindVersion(kind="dsse", version="0.0.1")
+        body_entry = TypeAdapter(ProposedEntry).validate_json(
+            tlog_entry.canonicalized_body
+        )
+        if not isinstance(body_entry, Hashedrekord) and not isinstance(
+            body_entry, Dsse
+        ):
```

Comment:
`isinstance` does a logical OR when given multiple types in a tuple 🙂 

```suggestion
        if not isinstance(body_entry, (Hashedrekord, Dsse)):
```